tests/runtest.proj

```diff
@@ -264,7 +268,7 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
 
                   System.IO.Directory.CreateDirectory(_Global.reportBase + testSubfolder)%3B
 
-                  ret = wrapper.RunTest(testExecutable, outputFile, errorFile)%3B
+                  ret = wrapper.RunTest(testExecutable, outputFile, errorFile, "$(TestArguments)")%3B
```

Comment:
I do not think that the tests should have arguments. The plan that we have been so far on was that the test are self-contained units.

There is prior art to what you are doing, e.g. with crossgen or with compiling tests as /debug- vs. /debug+. We have been adding these special execution modes as either:
- Global properties: set a global variable and run all tests in special execution mode
- Test local properties: special execution mode is specified in the test project file

I expect that we will be adding more of these special execution modes in future. If we keep driving them using global lists on the side, it will become extremely hard to reason about what the test does - one would need to look at the test project file, and also at a bunch of other global lists to figure out what special execution modes are turned on for it. It is not a good place to be in. We need to be in a place where you can tell what the test does by just looking at its project file.